runtime/queries/koka/highlights.scm

```diff
@@ -0,0 +1,237 @@
+; Function calls
+
+(appexpr
+  function: (appexpr
+    (atom
+      (qidentifier
+        [
+          (qvarid)
+          (qidop)
+          (identifier
+            [(varid) (idop)] @function)
+        ] @function))))
+
+(ntlappexpr
+  function: (ntlappexpr
+    (atom
+      (qidentifier
+        [
+          (qvarid)
+          (qidop)
+          (identifier
+            [(varid) (idop)] @function)
+        ] @function))))
+
+(appexpr
+  field: (atom
+    (qidentifier
+      [
+        (qvarid)
+        (qidop)
+        (identifier
+          [(varid) (idop)] @function)
+      ] @function)))
+
+(ntlappexpr
+  field: (atom
+    (qidentifier
+      [
+        (qvarid)
+        (qidop)
+        (identifier
+          [(varid) (idop)] @function)
+      ] @function)))
+
+[
+  "initially"
+  "finally"
+] @function.special
+
+; Function definitions
+
+(puredecl
+  (funid
+    (identifier
+      [(varid) (idop)] @function)))
+
+(fundecl
+  (funid
+    (identifier
+      [(varid) (idop)] @function)))
+
+(operation
+  (identifier
+    [(varid) (idop)] @function))
+
+; Identifiers
+
+(puredecl
+  (binder
+    (identifier
+      [(varid) (idop)] @constant)))
+
+; TODO: Highlight vars differently once helix has an appropriate highlight query
+; for that purpose.
+
+(pparameter
+  (pattern
+    (identifier
+      (varid) @variable.parameter)))
+
+(paramid
+  (identifier
+    (varid) @variable.parameter))
+
+(typeid
+  (varid) @type)
+
+(tbinder
+  (varid) @type)
+
+(typecon
+  (varid) @type)
+
+(qvarid
+  (qid) @namespace)
+
+(modulepath (varid) @namespace)
+
+(qconid) @namespace
+
+(qidop) @namespace
+
+(varid) @variable
+
+(conid) @constructor
+
+; Operators
+
+[
+  "!"
+  "~"
+  "="
+  ":="
+  (idop)
+  (op)
+  (qidop)
+] @operator
+
+; Keywords
+
+[
+  "as"
+  "behind"
+  (externtarget)
+  "forall"
+  "handle"
+  "handler"
+  "in"
+  "infix"
+  "infixl"
+  "infixr"
+  "mask"
+  "other"
+  "pub"
+  "some"
+] @keyword
```

Comment:
I don't think there's anything we can do about them. We can't predict how they'll be used, so we can't include them in the grammar. And if they're not in the grammar, you can't include them in the highlights, since they're not a valid query.

If there is a way that you know of though, let me know.